### PR TITLE
New component/location

### PR DESCRIPTION
**Pull request by Lelith**
**Who**: @lelith & @saibs 

**What / Why**: Create higher order component for LocationPicker / Autosuggest Element. This is based on http://react-autosuggest.js.org/

See the storybook demo about how you can use this higher order component.

**Further enhancements**: right now it uses not the TextInput Component as there was still some weird error with focus() when clicking on one of the suggestion

### CODE REVIEW

File: packages/components-style/src/components/form/_form_component.scss [hunk 1]
```diff
@@ -1,11 +1,34 @@
...
   line-height: 1em;
   padding: 10px;
+  width: 100%;
+  &:focus {
+    box-shadow: inset 0 0 5px rgba(0,0,0,.05);
+    outline: none;
+  }
+}
+
+*::-webkit-input-placeholder {
```
Comment (saibs) [inline packages/components-style/src/components/form/_form_component.scss]: Please check if autoprefixer can work here.

File: packages/components-style/src/components/form/_locationpicker.scss [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      fill: currentColor;
+    }
+  }
+
+  &__input {
+    padding-left: 45px;
+  }
+}
+
+.react-autosuggest{
```
Comment (saibs) [inline packages/components-style/src/components/form/_locationpicker.scss]: Maybe we can dedicated class and not changing global react-autosuggest.

Comment (Lelith) [reply packages/components-style/src/components/form/_locationpicker.scss]: I tried to use the [https://github.com/moroshko/react-autosuggest#renderSuggestionsContainerProp] function but i was not able to make it work. for some unknown reason it just ignored this function and did render always with the default.

Also we currently have a warning as the Demo Component is not providing a renderSuggestions function but this is required. It is, of course, bubbling up to the Higher order component and so it still works, but its also not nice to have this kind of warning, but i don t know how to fix it, without having a unnecessary dublication of this function in the demo component.

Comment (Lelith) [reply packages/components-style/src/components/form/_locationpicker.scss]: i fixed the warning.

**Comment by saibs** [comment]
LGTM.

**Pull request merged**